lib/output.js

```diff
@@ -409,7 +409,7 @@ function OutputStream(options) {
             print(encoded);
         },
         print_template_string_chars: function(str) {
-            var encoded = encode_string(str, '`');
+            var encoded = encode_string(str, '`').replace(/\${/g, "\\${");
```

Comment:
Uh yeah, watching again, this would require changes on 2 places (the regexp and the handling of the code). https://github.com/kzc/UglifyJS2/blob/1906d59dc47f80e6eb6c36f04855a10741982f02/lib/output.js#L136